swarm/network/stream/snapshot_retrieval_test.go

```diff
@@ -66,6 +68,44 @@ func TestFileRetrieval(t *testing.T) {
 	}
 }
 
+// TestPureRetrieval tests pure retrieval without syncing
+// A configurable number of nodes and chunks
+// can be provided to the test.
+// A number of random chunks is generated, then stored directly in
+// each node's localstore according to their address.
+// Each chunk is supposed to end up at certain nodes
+// With retrieval we then make sure that every node can actually retrieve
+// the chunks.
+func TestPureRetrieval(t *testing.T) {
+	var nodeCount []int
+	var chunkCount []int
+
+	if *nodes != 0 && *chunks != 0 {
+		nodeCount = []int{*nodes}
+		chunkCount = []int{*chunks}
+	} else {
+		nodeCount = []int{16}
+		chunkCount = []int{16}
+
+		if *longrunning {
+			nodeCount = append(nodeCount, 32, 64)
+			chunkCount = append(chunkCount, 32, 256)
+		} else if testutil.RaceEnabled {
+			nodeCount = []int{4}
+			chunkCount = []int{4}
```

Comment:
Same comment here... too much orchestration happening for spinning up 4 nodes and connecting them, only to store and check 4 chunks.

Reply:
The reason has always been that it took very long for these simulations to run.
This specific number of 4 has been added by @frncmx and is only relevant for race enabled tests, otherwise apparently these tests were always incurring in race problems

Reply:
That seems rather odd - if the test indeed doesn't work with more than a few chunks, this is really worrying.

However I reviewed these tests while working on another PR about 1-2 weeks ago, and they worked just fine with 40, or 400 chunks.

So I suggest we bump up from 4 to at least 40-50 chunks (just so that we make sure at least 1 chunk ends up on every node)... probably due to speed on Travis it is not a good idea to run with more than 100-200 chunks (although this is not a lot in any case either), but 4 is ridiculously low for such a test.

If this test doesn't work with 40-50 chunks, I fail to see its value altogether...

Reply:
I've been under the impression that the slowness comes from the fact that we start many nodes, which in term open many peer connections, start many subscriptions, etc.

So it is kinda expected for these tests to be very slow with a high number of Swarm nodes, but I don't think the number of chunks should really have such an effect.

Reply:
Agreed. High chunk number should have less impact than high nodes number. 

However again take note that the very low number of 4 is for race-enabled testing only, for normal unit testing we use ` chnkCnt = []int{4, 32, 256}` for `TestRetrieval`, ` chunkCount = []int{16}` for `TestPureRetrieval`, and ` (rand.Intn(maxFileSize-minFileSize) + minFileSize) *1024` for `TestFileRetrieval` (through `generateRandomFile()`.